vm.go

```diff
@@ -381,6 +469,7 @@ func (vm *vm) run() {
 			atomic.StoreUint32(&vm.interrupted, 0)
 
 		} else {
+			// TODO: vm.pc can be -1 which would cause this to panic
 			vm.prg.code[vm.pc].exec(vm)
```

Comment:
```go
vm.prg.code[vm.pc].exec(vm)
```

but `vm.pc` can be `-1` which would cause a panic.

Reply:
This -1 is also baked into goja (on their master).  which test case is causing this to panic?

Reply:
I'm about to look into it a bit more but it's `TestIteratorReturnErrorNested`

Reply:
I looked into it Friday but didn't get to isolate under which condition -1 is triggered. I might be able to investigate again soon.